GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -698,6 +699,61 @@ public GVRSceneObject getChildByIndex(int index) {
      */
     public Iterable<GVRSceneObject> children() {
         return new Children(this);
+    }    
+
+    /**
+     * Add a {@link GVRBaseSensor} node to the object.
+     * 
+     * Note that the attached sensor affects this node and its descendants.
+     * 
+     * @param sensor
+     *            The {@link GVRBaseSensor} to be added to the object node.
+     *            
+     * @return <code>true</code> if the sensor was added successfully;
+     *         <code>false</code> otherwise, this could possibly be due to the
+     *         {@link GVRInputManager} not being set.
+     */
+    public boolean addSensor(GVRBaseSensor sensor) {
```

Comment:
Ok, these additions do not seem to me to belong "organically" to a GVRSceneObject. GVRSceneObject has numerous APIs the way it is. This "augmentation" of GVRSceneObject instances can easily be encapsulated by the GVRInputManager or SensorManager. Either one of these could associate a GVRSceneObject with a Sensor so later the SensorManager can lookup a Sensor for a SceneObject (referring to SensorManager.recurseSceneObject which uses GVRSceneObject.getSensor). Let's try to avoid making knowledge of this input/sensor concept too wide.